dds/DCPS/XTypes/TypeLookupService.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_TYPE_LOOKUP_SERVICE_H
+#define OPENDDS_DCPS_TYPE_LOOKUP_SERVICE_H
+
+#include "TypeObject.h"
+
+#include <dds/DCPS/RcObject.h>
+#include <dds/DCPS/SequenceNumber.h>
+#include <dds/DCPS/TypeSupportImpl.h>
+
+#if !defined (ACE_LACKS_PRAGMA_ONCE)
+#pragma once
+#endif /* ACE_LACKS_PRAGMA_ONCE */
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+class OpenDDS_Dcps_Export TypeLookupService : public virtual DCPS::RcObject {
+public:
+  TypeLookupService();
+  ~TypeLookupService();
+
+  // For TypeAssignability
+  const TypeObject& get_type_objects(const TypeIdentifier& type_id) const;
+  void add_type_objects_to_cache(const TypeIdentifier& ti, const TypeObject& tobj);
+
+  // For Type Lookup Service request / reply
+  void get_type_objects(const TypeIdentifierSeq& type_ids,
+    TypeIdentifierTypeObjectPairSeq& types);
+  void add_type_objects_to_cache(TypeIdentifierTypeObjectPairSeq& types);
+
+  // For adding local endpoint types
+  void add_type_objects_to_cache(const DCPS::TypeSupportImpl& typesupport);
+
+  bool type_object_in_cache(const TypeIdentifier& ti);
+
+  OpenDDS::DCPS::SequenceNumber rpc_sequence_number_;
```

Comment:
Does this need to be public?
Also, don't add `OpenDDS::` qualification

Reply:
It has to be public the way code is using it now. Do you want me to change it to private with access functions?

Reply:
Made private and added public API